lib/suse/connect/client.rb

```diff
@@ -124,7 +124,7 @@ def announce_system(distro_target = nil, instance_data_file = nil)
 
         response = @api.announce_system(*params)
 
-        [response.body['login'], response.body['password']]
```

Comment:
It shouldn't be a problem because if the client assumes a tuple, then the following code would happen (as it happens internally for connect):

```ruby
login, password = ['login', 'password', 'token']
# => login = 'login'
# => password = 'password'
```

That being said, we never know if the client will perform some weird check like ensuring a given size for the return value of `announce_system`. Thus, one thing I can do is to force this on `lib/suse/connect/yast.rb`.

Reply:
Ok, nevermind that. Since `announce_system` is using the token-based authentication, the `system-token` mechanism would not kick in yet. Thus, it's safe to assume that we can revert back this part and the `system-token` mechanism will kick in on the first system-based authentication.